File: LeetCode/Simulation/Golang/Leetcode929-unique-email-addresses.go
package sbgo

type void struct{}

func numUniqueEmails(emails []string) int {
	st := map[string]void{}
	for _, s := range emails {
		t := ""
		for i := 0; i < len(s); i++ {
			if s[i] == '.' {
				continue
			}
			if s[i] == '@' {
				t += s[i:]
				break
			}
			if s[i] == '+' {
				for s[i] != '@' {
					i++
				}
				t += s[i:]
				break
			}
			t += s[i : i+1]
		}
		st[t] = void{}
	}
	return len(st)
}

File: LeetCode/Simulation/Python/Leetcode929-unique-email-addresses.py
class Solution:
    def numUniqueEmails(self, emails: List[str]) -> int:
        st = set()
        for s in emails:
            t = []
            for i in range(len(s)):
                if s[i] == '.':
                    continue
                if s[i] == '+':
                    while s[i] != '@':
                        i += 1
                    t.append(s[i:])
                    break
                if s[i] == '@':
                    t.append(s[i:])
                    break
                t.append(s[i])
            st.add(''.join(t))
        return len(st)
